javacutil/src/org/checkerframework/javacutil/AnnotationUtils.java

```diff
@@ -377,6 +381,92 @@ public static boolean hasInheritedMeta(AnnotationMirror anno) {
     }
 
     /**
+     * Returns true if the two annotations have the same elements (fields). The arguments {@code
+     * am1} and {@code am2} must be the same type of annotation.
+     */
+    public static boolean sameElementValues(AnnotationMirror am1, AnnotationMirror am2) {
+        Map<? extends ExecutableElement, ? extends AnnotationValue> vals1 = am1.getElementValues();
+        Map<? extends ExecutableElement, ? extends AnnotationValue> vals2 = am2.getElementValues();
+        for (ExecutableElement meth :
+                ElementFilter.methodsIn(
+                        am1.getAnnotationType().asElement().getEnclosedElements())) {
+            AnnotationValue aval1 = vals1.get(meth);
+            AnnotationValue aval2 = vals2.get(meth);
+            if (aval1 == null) {
+                aval1 = meth.getDefaultValue();
+            }
+            if (aval2 == null) {
+                aval2 = meth.getDefaultValue();
+            }
+            if (!sameAnnotationValue(aval1, aval2)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Return true iff the two AnnotationValue objects are the same. Use this instead of
+     * CheckerFrameworkAnnotationValue.equals, which wouldn't get called if the receiver is some
+     * AnnotationValue other than CheckerFrameworkAnnotationValue.
+     */
+    public static boolean sameAnnotationValue(AnnotationValue av1, AnnotationValue av2) {
+        if (av1 == av2) {
+            return true;
+        }
+        if (av1 == null || av2 == null) {
+            return false;
+        }
+        return sameAnnotationValueValue(av1.getValue(), av2.getValue());
+    }
+
+    /**
+     * Return true if the two annotation values are the same. The arguments to this method are
+     * values that are returned by {@code AnnotationValue.getValue()}.
+     */
+    public static boolean sameAnnotationValueValue(Object val1, Object val2) {
```

Comment:
Should this be a `private` helper method?

Reply:
I was thinking it might possibly be useful to a caller who didn't want to use `sameAnnotationValue` for some reason.  We could make it `private` if you feel that it pollutes the `AnnotationUtils` interface.